T2CC_APP/t2cc/app/src/main/java/com/example/t2cc/RegistrationActivity.java

```diff
@@ -227,7 +243,32 @@ private boolean validateRegistrationForm() {
       mPasswordField.setError("Required.");
       mPasswordField.requestFocus();
       valid = false;
-    } else {
+    } else if (password.length() < 10) {
+      mPasswordField.setError("Password needs to be at least 10 char. ");
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containNumber(password)){
+      mPasswordField.setError("Password need to contain 1 number");
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containLowerCase(password)){
+      mPasswordField.setError("Password need to contain 1 lower case");
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if (!containUpperCase(password)){
+      mPasswordField.setError("Password need to contain 1 Upper case");
+      mPasswordField.requestFocus();
+      valid = false;
+    }
+    else if(!containSpecialChar(password)){
+      mPasswordField.setError("Password need to contain 1 special characters (!@#$%^&*())");
```

Comment:
Could we have these requirements (all of these for the PW in teeny tiny text under the pw, so they don't have to keep trying and failing?)